### PR TITLE
feat(transactions): send_7702_tx

### LINKED ISSUE #128: Add EIP-7702 example: `send_eip7702_transaction.rs`
EIP-7702 adds a new tx type (`SET_CODE_TX`) which contains a new list: `authorization_list`. Each item in this list is a tuple of `(chain_id, address, nonce)` which is signed. Since this is an additional signing flow, it might be a bit hard to figure out how to use it.

One important distinction is that the items in the `authorization_list` are the actual accounts affected: i.e., the accounts "abstracted" in this AA proposal are the signers of each individual item. This means that the signatory of the main transaction is actually a sponsor.

It is possible a nice example here requires some additional helpers in Alloy for working with authorization lists.

See the EIP for more details: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7702.md

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Closes #128 

## Solution

send 7702 tx. Need latest anvil and alloy

## PR Checklist

- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: examples/transactions/examples/send_eip7702_transaction.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    // Sign and Encode the transaction
+    let sig = sender.sign_transaction_sync(&mut tx)?;
+    let tx = tx.into_signed(sig);
+    let mut encoded = Vec::new();
+    tx.tx().encode_with_signature(tx.signature(), &mut encoded, false);
+    let receipt = provider.send_raw_transaction(&encoded).await?.get_receipt().await?;
+
+    assert!(receipt.status());
+    assert!(receipt.inner.logs().len() == 1);
```
Comment (mattsse) [inline examples/transactions/examples/send_eip7702_transaction.rs:85]: can use assert_eq!

File: examples/transactions/examples/send_eip7702_transaction.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    assert!(receipt.inner.logs()[0].address() == authority.address());
```
Comment (mattsse) [inline examples/transactions/examples/send_eip7702_transaction.rs:86]: same

**Review by mattsse — CHANGES_REQUESTED** [review]
pedantic nit, can merge after new alloy release

**Review by zerosnacks — APPROVED** [review]
PR lgtm, I don't think we would want to merge this into main with the patch and instead wait for the next Alloy release?

File: Cargo.toml [hunk 3]
```diff
@@ -117,3 +117,6 @@ tokio = "1.38"
 eyre = "0.6"
 serde = "1.0"
 serde_json = "1.0"
+
+[patch.crates-io]
+alloy = { git = "https://github.com/alloy-rs/alloy", rev = "f5eaddf" }
```
Comment (zerosnacks) [inline Cargo.toml:122]: TODO: remove

**Review by onbjerg — APPROVED** [review]

File: examples/transactions/examples/send_eip7702_transaction.rs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    providers::{Provider, ProviderBuilder},
+    signers::{local::LocalSigner, SignerSync},
+    sol,
+};
+use eyre::Result;
+
+// Codegen from embedded Solidity code and precompiled bytecode.
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc, bytecode = "6080604052348015600f57600080fd5b5060d980601d6000396000f3fe6080604052348015600f57600080fd5b506004361060325760003560e01c80637b3ab2d01460375780639ee1a44014603f575b600080fd5b603d6047565b005b60456075565b005b7fbcdfe0d5b27dd186282e187525415c57ea3077c34efb39148111e4d342e7ab0e60405160405180910390a1565b7f2d67bb91f17bca05af6764ab411e86f4ddf757adb89fcec59a7d21c525d4171260405160405180910390a156fea2646970667358221220eff0a4721c5625c372d3b4a7e348d36d5ee4dd32a26abdd94cba7839b45852d764736f6c63430008190033")]
```
Comment (DaniPopes) [inline examples/transactions/examples/send_eip7702_transaction.rs:17]: can you add the command you used to compile this

**Comment by yash-atreya** [comment]
Awaiting new alloy release before merging

**Comment by onbjerg** [comment]
@yash-atreya alloy 0.3.0 is released, so this should be unblocked

**Pull request merged**